hazelcast/src/main/java/com/hazelcast/query/impl/Indexes.java

```diff
@@ -57,6 +57,9 @@
     public static final int SKIP_PARTITIONS_COUNT_CHECK = -1;
     private static final InternalIndex[] EMPTY_INDEXES = {};
 
+    private static final ThreadLocal<CachedQueryEntry[]> CACHED_ENTRIES =
```

Comment:
is there a chance these thread locals are not cleaned up -> there are live references to `SerializationService` and `Extractors` of an already shutdown HZ instance that cannot be collected?

Reply:
Yes, I think that's possible, but I'm not sure what we can do about that.

Reply:
>Or probably that's not possible :) If there are no partition threads alive after the shutdown, the thread locals should be automatically cleared sooner or later.

@taburet can `QueryCache` be an exception here (given that user thread access is possible)?

Reply:
Yes, it can be.